src/bmf.cpp

```diff
@@ -141,48 +145,46 @@ bool CxadbmfPlayer::xadplayer_load()
   }
   else
   {
-    strncpy(bmf.title,xad.title,36);
-    strncpy(bmf.author,xad.author,36);
+    strncpy(bmf.title, xad.title, 36);
+    strncpy(bmf.author, xad.author, 36);
   }
 
   // speed
   if (tune_size - ptr < 1)
     return false;
-  if (bmf.version > BMF0_9B)
-    bmf.speed = tune[ptr++];
-  else
-    bmf.speed = ((tune[ptr++] << 8) / 3) >> 8; // strange, yeh ?
+  bmf.speed = tune[ptr++];
+  if (bmf.version == BMF0_9B)
+    bmf.speed /= 3; // removed pointless shifts from original code
 
   // load instruments
   if (bmf.version > BMF0_9B)
   {
     if (tune_size - ptr < 4)
       return false;
-    unsigned long iflags = (tune[ptr] << 24) | (tune[ptr+1] << 16) | (tune[ptr+2] << 8) | tune[ptr+3];
-    ptr+=4;
+    unsigned long iflags = (tune[ptr] << 24) | (tune[ptr+1] << 16)
+                         | (tune[ptr+2] << 8) | tune[ptr+3];
+    ptr += 4;
 
-    for(i=0;i<32;i++)
+    for (i = 0; i < 32; i++)
+    {
       if (iflags & (1 << (31-i)))
-	  {
+      {
         if (tune_size - ptr < 24)
           return false;
         strncpy(bmf.instruments[i].name, (char *)tune + ptr, 10);
         // 11th char ignored since name must be 0-terminated
         bmf.instruments[i].name[10] = 0;
-        memcpy(bmf.instruments[i].data, &tune[ptr+11], 13);
+        memcpy(bmf.instruments[i].data, tune + ptr + 11, 13);
```

Comment:
Same argument here.  Given that `[]` is used a lot elsewhere, I think keeping `[]` in use makes the code a bit more uniform.  Otherwise you could make the argument that on line 161 above, `(tune[ptr] << 24) | (tune[ptr + 1] << 16)` should be replaced with the horrible looking `(*(tune + ptr) << 24) | (*(tune + ptr + 1) << 16))` to match the new change here :)

Reply:
See above. But to be clear, I don't want to ban `[]`, I just don't like `&[]`. No sane person would write `*(tune + ptr)`, and I've even changed some occurrences of `*(stream+1)` to `stream[1]` further down (although I can see why one might prefer the other form in that particular contex; and now `stream[1]` is used next to `*stream`, which is inconsistent, but `*stream` is a better fit than `stream[0]` there; whatever you do, you have to live with one ugly wart). Some people want conistency, others use different syntax depending on context to emphasize certain aspects, it's all complicated. I don't have a very strong opinion on this matter, just tell me what you want.

Reply:
Generally my opinion is that `&[]` is clearer to me when talking about offsets in files, as it's obvious you're looking at an offset because of the array syntax.  Something like `tune + 11` requires more mental effort for me, as I have to remember whether `tune` is being used as a pointer or an integer.